core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/FileSystemResourcesBuildStep.java

```diff
@@ -7,22 +7,41 @@
 import java.nio.file.Path;
 import java.util.List;
 
+import io.quarkus.deployment.IsNormal;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.GeneratedFileSystemResourceBuildItem;
+import io.quarkus.deployment.builditem.GeneratedFileSystemResourceHandledBuildItem;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
 import io.quarkus.deployment.pkg.builditem.ArtifactResultBuildItem;
 import io.quarkus.deployment.pkg.builditem.OutputTargetBuildItem;
+import io.quarkus.runtime.LaunchMode;
 
 public class FileSystemResourcesBuildStep {
 
-    @BuildStep
-    public void write(OutputTargetBuildItem outputTargetBuildItem,
+    @BuildStep(onlyIfNot = IsNormal.class)
+    public void notNormalMode(OutputTargetBuildItem outputTargetBuildItem,
+            LaunchModeBuildItem launchMode,
+            List<GeneratedFileSystemResourceBuildItem> generatedFileSystemResources,
+            BuildProducer<GeneratedFileSystemResourceHandledBuildItem> producer) {
+        if (launchMode.getLaunchMode() == LaunchMode.DEVELOPMENT) {
```

Comment:
I don't think you need these changes if you remove GeneratedFileSystemResourceHandledBuildItem

Reply:
That doesn't work, because then `FileSystemResourcesBuildStep` doesn't get the chance to do the actual writing of the resources.
That means that the Kubernetes Process does run and it does create the proper `FileSystemResourcesBuildStep` objects, but no code takes care of writing them.

We could move the code that writes the result outside of the build, but I don't like it that much.

WDYT?

Reply:
Ah, I see why it needed now. Sorry for the noise

Reply:
Np, thanks for checking :)